uamqp/cbs.py

```diff
@@ -203,10 +204,20 @@ def handle_token(self):
             self.update_token()
             return False
         elif self.auth_state == CbsAuthState.Failure:
-            raise AuthenticationException("Failed to open CBS authentication link.")
+            raise AuthenticationException(
+                condition=ErrorCodes.InternalError,
+                description="Failed to open CBS authentication link."
+            )
         elif self.auth_state == CbsAuthState.Error:
-            raise TokenAuthFailure(self._token_status_code, self._token_status_description, self._encoding)
+            raise TokenAuthFailure(
+                self._token_status_code,
+                self._token_status_description,
+                encoding=self._encoding
```

Comment:
I'm pretty sure we can strip out all the _encoding stuff at some point.

Reply:
left a todo comment